File: ios/Modules/OpenWrap/OpenWrap.m
//
//  OpenWrap.m
//  OpenWrapInRN
//
//  Created by Zhengqian Kuang on 2022-03-27.
//

#import <Foundation/Foundation.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(GBOpenWrap, NSObject)

RCT_EXTERN_METHOD(initializeSDK:(NSString)storeURLString)

@end

File: ios/Modules/OpenWrap/OpenWrap.swift
//
//  OpenWrap.swift
//  OpenWrapInRN
//
//  Created by Zhengqian Kuang on 2022-03-27.
//

import Foundation
import OpenWrapSDK

@objc(OpenWrap)
class OpenWrap: NSObject {

  @objc static func requiresMainQueueSetup() -> Bool {
    return false;
  }

  @objc(initializeSDK:)
  func
  initializeSDK(storeURLString: String) {
    if let storeURL = URL(string: storeURLString) {
      let appInfo = POBApplicationInfo()
      appInfo.storeURL = storeURL
      OpenWrapSDK.setApplicationInfo(appInfo)
    }
    
    #if DEBUG
    OpenWrapSDK.setLogLevel(.all)
    #endif
  }

}
